internal/jobs/launch_instance_aws.go

```diff
@@ -143,28 +143,40 @@ func DoEnsurePubkeyOnAWS(ctx context.Context, args *LaunchInstanceAWSTaskArgs) e
 	// check presence on AWS first
 	fingerprint := pubkey.FindAwsFingerprint(ctx)
 	ec2Name, err := ec2Client.GetPubkeyName(ctx, fingerprint)
-	if err != nil {
-		span.SetStatus(codes.Error, "key error")
+
+	if errors.Is(err, http.PubkeyNotFoundErr) {
+		logger.Trace().Msgf("Pubkey with fingerprint '%s' not found, importing type %s key '%s'", fingerprint, pubkey.Type, pubkey.Body)
 
 		// if not found on AWS, import
-		if errors.Is(err, http.PubkeyNotFoundErr) {
-			pkr.Tag = ""
-			pkr.RandomizeTag()
-			pkr.Handle, err = ec2Client.ImportPubkey(ctx, pubkey, pkr.FormattedTag())
-
-			if errors.Is(err, http.DuplicatePubkeyErr) {
-				// key not found by fingerprint but importing failed for duplicate err so fingerprints do not match
-				return fmt.Errorf("key with fingerprint %s not found on AWS, but importing the key failed: %w", pubkey.Fingerprint, err)
-			} else if err != nil {
-				return fmt.Errorf("cannot upload aws pubkey: %w", err)
+		pkr.Tag = ""
+		pkr.RandomizeTag()
+		var retryCount int
+		origName := pubkey.Name
+	retry:
+		pkr.Handle, err = ec2Client.ImportPubkey(ctx, pubkey, pkr.FormattedTag())
+
+		if errors.Is(err, http.DuplicatePubkeyErr) {
+			retryCount += 1
+			pubkey.Name = fmt.Sprintf("%s%d", origName, retryCount+1)
```

Comment:
Just an idea: add a separator between the key name and new suffix. If we had a key named "key1", it suddenly becomes a "key12" if we upload new. But that is nothing major. :thinking:

Reply:
I don’t thing we ever want to retry more than 3-5 times. If you insist I can rebase, this one looks more "human" tho.

Reply:
No, I think that the amount of retries is sufficient. I was overthinking the suffix appending. I think this is okay. :+1: